source/extensions/transport_sockets/http_11_proxy/config.cc

```diff
@@ -0,0 +1,43 @@
+#include "source/extensions/transport_sockets/http_11_proxy/config.h"
+
+#include "envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.pb.h"
+#include "envoy/extensions/transport_sockets/http_11_proxy/v3/upstream_http_11_connect.pb.validate.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/config/utility.h"
+#include "source/extensions/transport_sockets/http_11_proxy/connect.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace TransportSockets {
+namespace Http11Connect {
+
+Network::TransportSocketFactoryPtr
+UpstreamHttp11ConnectSocketConfigFactory::createTransportSocketFactory(
+    const Protobuf::Message& message,
+    Server::Configuration::TransportSocketFactoryContext& context) {
+  const auto& outer_config = MessageUtil::downcastAndValidate<
+      const envoy::extensions::transport_sockets::http_11_proxy::v3::Http11ProxyUpstreamTransport&>(
+      message, context.messageValidationVisitor());
+  auto& inner_config_factory = Config::Utility::getAndCheckFactory<
+      Server::Configuration::UpstreamTransportSocketConfigFactory>(outer_config.transport_socket());
+  ProtobufTypes::MessagePtr inner_factory_config = Config::Utility::translateToFactoryConfig(
```

Comment:
LOL, why is this not auto? :) (But I'm happy with explicit types so don't ever change! :>)